<p style="background-color:mediumseagreen;font-family:newtimeroman;font-size:200%;color:white;text-align:center;border-radius:60px 20px;"><b>Computer Vision - YOLOv8</b></p>

[ultralytics GitHub Link](https://github.com/ultralytics/ultralytics)

In [1]:
# pip install opencv-python

In [ ]:
# pip install ultralytics           

YOLO has the ability to detect objects by processing the image only once. ("You Only Look Once")

The image is divided into a grid with a predefined scale. Each grid cell contains a certain number of bounding boxes.

Each bounding box is used to predict the class and location of objects that may be present in that region. These predictions are made through a deep convolutional neural network (CNN).

As a result, YOLO can detect multiple objects simultaneously and provides these detections as an output.

In summary, YOLO's unique working principle involves dividing the image into grids and using the bounding boxes in each grid cell for object detection. On the other hand, CNNs are generally used to extract features from an image dataset and perform various tasks using these features. YOLO is an object detection model that incorporates CNN-based layers.

In [1]:
# Those who get errors can try this code
import cv2

import numpy as np
import matplotlib.pyplot as plt

**Note:** You should close the kernel to turn off the camera.

In [3]:
from ultralytics import YOLO
#from ultralytics.yolo.v8.detect.predict import DetectionPredictor   
from ultralytics.models.yolo.detect import DetectionPredictor 

In [5]:
# This code block opens your computer camera and detects objects.

# Load a model
model = YOLO("yolov8n.pt")            # load a pretrained model (recommended for training)

# Use the model

results= model.predict(source="0", show=True)
print(results)


1/1: 0... Success  (inf frames of shape 640x480 at 30.00 FPS)


WARNING  inference results will accumulate in RAM unless `stream=True` is passed, causing potential out-of-memory
errors for large sources or long-running streams and videos. See https://docs.ultralytics.com/modes/predict/ for help.

Example:
    results = model(source=..., stream=True)  # generator of Results objects
    for r in results:
        boxes = r.boxes  # Boxes object for bbox outputs
        masks = r.masks  # Masks object for segment masks outputs
        probs = r.probs  # Class probabilities for classification outputs

0: 480x640 2 persons, 98.0ms
0: 480x640 1 person, 82.1ms
0: 480x640 1 person, 81.2ms
0: 480x640 1 person, 165.1ms
0: 480x640 1 person, 84.1ms
0: 480x640 1 person, 75.4ms
0: 480x640 1 person, 70.0ms
0: 480x640 1 person, 65.0ms
0: 480x640 1 person, 71.0ms
0: 480x640 1 person, 73.3ms
0: 480x640 1 person, 65.5ms
0: 480x640 1 person, 75.4ms
0: 480x640 1 person, 67.7ms
0: 480x640 1 person, 63.3ms
0

KeyboardInterrupt: 

In [5]:
model = YOLO("yolov8n.pt")            # load a pretrained model (recommended for training)
result_1 = model.predict("dog.png")


image 1/1 C:\Users\mertu\Masast\Batch_270\3-Computer Vission\YoloV8_b270\dog.png: 448x640 1 dog, 129.4ms
Speed: 0.0ms preprocess, 129.4ms inference, 1199.7ms postprocess per image at shape (1, 3, 448, 640)


In [6]:
model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)
result_2 = model.predict("sky.jpg")


image 1/1 C:\Users\mertu\Masast\Batch_270\3-Computer Vission\YoloV8_b270\sky.jpg: 384x640 1 airplane, 94.0ms
Speed: 15.8ms preprocess, 94.0ms inference, 0.0ms postprocess per image at shape (1, 3, 384, 640)


In [7]:
model = YOLO("yolov8n.pt")  # load a pretrained model (recommended for training)
results = model.predict(source="smooth_traffic.mp4")  



WARNING  inference results will accumulate in RAM unless `stream=True` is passed, causing potential out-of-memory
errors for large sources or long-running streams and videos. See https://docs.ultralytics.com/modes/predict/ for help.

Example:
    results = model(source=..., stream=True)  # generator of Results objects
    for r in results:
        boxes = r.boxes  # Boxes object for bbox outputs
        masks = r.masks  # Masks object for segment masks outputs
        probs = r.probs  # Class probabilities for classification outputs

video 1/1 (frame 1/166) C:\Users\mertu\Masast\Batch_270\3-Computer Vission\YoloV8_b270\smooth_traffic.mp4: 640x384 9 cars, 1 bus, 1 truck, 113.0ms
video 1/1 (frame 2/166) C:\Users\mertu\Masast\Batch_270\3-Computer Vission\YoloV8_b270\smooth_traffic.mp4: 640x384 9 cars, 2 buss, 1 truck, 91.0ms
video 1/1 (frame 3/166) C:\Users\mertu\Masast\Batch_270\3-Computer Vission\YoloV8_b270\smooth_traffic.mp4: 640x384 1 person, 9 cars, 2 buss, 88.1ms
video 1/1 (frame

The number of frames in a video corresponds to a certain duration in seconds, depending on the video's frame rate (FPS). FPS represents the number of frames displayed per second in a video. For example, if a video has a frame rate of 30 FPS, it means 30 frames are displayed every second.

Without knowing the frame rate, it is difficult to determine exactly how many seconds 166 frames correspond to. However, standard frame rates are commonly used in videos. For example, if we calculate based on the widely used 30 FPS:

166 / 30 = 5.53 sec

<p style="background-color:mediumseagreen;font-family:newtimeroman;font-size:200%;color:white;text-align:center;border-radius:60px 20px;"><b>THANK YOU!</b></p>